### PR TITLE
Reworked how blocker data is displayed in reports and reminders

### LINKED ISSUE #140: "No blocker" annotations are not very helpful when some blockers exist
This is an opinion rather than a bug. I feel we could make reporting compact by not emitting extra rows for bugzillas/tickets/others if there aren't any. For example, here's a job that has no bugzillas or tickets, but has a blocker annotation:

Screenshot 2020-07-22 09 30 15

It would be better if we could represent it like this:

Screenshot 2020-07-22 09 33 02

When we have a failing job that has no blockers of any kind defined, we could call that out in a single line:

_"No blockers have been filed for this job"_

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Fixes #140 

This is a larger change, so here are some high-level details:
- As per the README, the integer value of 0 is used as a "placeholder" value in `blockers.yaml` for jobs that have no active blockers filed either as bugs or tickets.
- Previously, whether or not the value was present in a non-passing job was used to make a determination between a "triaged" job (failing but 0 value is present) and a "non-triaged" job (job not present in blockers file). This distinction was previously used in the reporting, but is no longer made.
- Rather, the logic has been retooled so that the 0 value is scrubbed from blocker files and rather a calculation is made as to whether or not a job has **any** blockers filed, and have adjusted the reporting accordingly. Note that 0 can and should still be used as a placeholder value for job blockers.
- Also made some minor changes in colors used for LCB result in both the chart and table

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
@reviewers I have left the print debug statements in for now, please test this with your DFGs jobs when you get the chance and ensure everything works as expected, including summary statistics

**Comment by agarcia64** [comment]
Tested the changes on my machine, everything looks good to me.

**Review by agarcia64 — APPROVED** [review]

File: report.py
Comment (itzikb) [inline report.py:75]: This should be changed to FAILURE

**Review by itzikb — CHANGES_REQUESTED** [review]

File: report.py
Comment (amodi5) [inline report.py:166]: Isn't Aborted color grey in jenkins?

Comment (nathan-weinberg) [reply report.py:166]: Nope, "Aborted" is black and "Not built" is gray

**Review by AlistairTonner — COMMENTED** [review]
pulled and ran on local system output includes "no build found" and includes blockers_hardprov.yaml entries for those no build found jobs.

I"m off to add "aborted CI run" for all the aborted jobs

**Review by AlistairTonner — APPROVED** [review]
Okay -- stil figuring out this interface.....

**Review by agarcia64 — APPROVED** [review]
Just tested the latest changes. Everything looks good on my end!

File: report.py
Comment (amodi5) [reply report.py:166]: ah yes, maybe it's changed with the new jenkins CI

**Comment by amodi5** [comment]
Tested it, the data and all look correct, there is a minor suggestion though. The color between the chart and the report seems to not match exactly, there is a +/- shade difference as shown in the screenshots below, can we fix it?

unnamed (2)

Screenshot from 2020-09-15 15-48-27

**Comment by nathan-weinberg** [comment]
@amodi5 I tried implementing it where I used hex values for colors instead of the generic color choices, but could not get it to work :(

Maybe we could make this part of @pavankrao's followup patch?

**Review by amodi5 — APPROVED** [review]

**Comment by nathan-weinberg** [comment]
@itzikb @jagee Please review latest changes as soon as is convenient, will merge if no issues are found

File: remind.py
Comment (timflannagan1) [inline remind.py:45]: I would recommend in future iterations tracking this state as a constant variable.

File: functions.py
Comment (timflannagan1) [inline functions.py:293]: You may want to move this logic into the actual loop when you're building up the set of Jira ticket IDs. That way, you avoid processing the set twice, but realistically, the overhead for making this call to discard a single element from a set shouldn't be that expensive.

**Comment by nathan-weinberg** [comment]
Going to merge this by EOD if no issues are reported

**Comment by gouthampacha** [comment]
Great stuff - thanks for this fix!

**Pull request merged**
